src/tox/util/spinner.py

```diff
@@ -0,0 +1,156 @@
+"""A minimal non-colored version of https://pypi.org/project/halo, to track list progress"""
+from __future__ import absolute_import, unicode_literals
+
+import os
+import sys
+import threading
+import time
+from datetime import datetime
+from threading import RLock
+
+threads = []
+
+if os.name == "nt":
+    import ctypes
+
+    class _CursorInfo(ctypes.Structure):
+        _fields_ = [("size", ctypes.c_int), ("visible", ctypes.c_byte)]
+
+
+class Spinner(object):
+    CLEAR_LINE = "\033[K"
+    refresh_rate = 0.1
+    max_width = 120
+    frames = ["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"]
+
+    def __init__(self, enabled=True):
+        self._lock = RLock()
+        self._envs = dict()
+        self._frame_index = 0
+        self.enabled = enabled
+        self.stream = sys.stdout
+
+    def add(self, name):
+        self._envs[name] = datetime.now()
+
+    def clear(self):
+        if self.enabled:
+            with self._lock:
+                self.stream.write("\r")
+                self.stream.write(self.CLEAR_LINE)
+
+    def render(self):
+        while not self._stop_spinner.is_set():
+            self.render_frame()
+            time.sleep(self.refresh_rate)
+        return self
+
+    def render_frame(self):
+        if self.enabled:
+            with self._lock:
+                self.clear()
+                self.stream.write("\r{}".format(self.frame()))
+
+    def frame(self):
+        frame = self.frames[self._frame_index]
+        self._frame_index += 1
+        self._frame_index = self._frame_index % len(self.frames)
+        text_frame = "[{}] {}".format(len(self._envs), " | ".join(self._envs))
+        if len(text_frame) > self.max_width - 1:
+            text_frame = "{}...".format(text_frame[: self.max_width - 1 - 3])
+        return "{} {}".format(*[(frame, text_frame)][0])
+
+    def __enter__(self):
+        if self.enabled:
+            self.disable_cursor()
+
+        self._stop_spinner = threading.Event()
+        self._spinner_thread = threading.Thread(target=self.render)
+        self._spinner_thread.setDaemon(True)
+        self._spinner_id = self._spinner_thread.name
+        self._spinner_thread.start()
+        return self
+
+    def succeed(self, key):
+        self.finalize(key, "✔ OK")
```

Comment:
I assume just fine as the CI passes 😄 I tested from within cmder too on Windows and displays just fine (similar to how black emojis doo). So for now I don't plan adding some fallback symbols, although we may need down the line.

Reply:
Heh, black's don't render for me on windows so I suspect these might suffer as well

Reply:
On `Cmder` it will work fine (which I use myself too), but on `cmd` things might be different... let me test this real quick.

Reply:
![image](https://user-images.githubusercontent.com/1085180/50787779-865ec000-129e-11e9-8ce0-0fa5aa6f6a3b.png)

😢

Reply:
On Cmder it looks fine 😁 

![image](https://user-images.githubusercontent.com/1085180/50788242-f15cc680-129f-11e9-8968-e440bb6f5694.png)

Can we make use colors for the symbols? It would improve the output quite a bit I believe

Reply:
Sure any colour suggestions? Is there a reliable way to detect if the terminal supports these unicode characters, so we can fallback to something ASCII?

Reply:
> Sure any colour suggestions?

Green for "success" and red for "failure". The environment names might also appear in blue for easier viewing.

> Is there a reliable way to detect if the terminal supports these unicode characters, so we can fallback to something ASCII?

On pytest we use `colorama` to handle the colors, I assumed `tox` was using the same thing.

Reply:
Nope, we only use the ``py.io.TerminalWriter``.